hie-base/Haskell/Ide/Engine/PluginTypes.hs

```diff
@@ -175,8 +179,51 @@ type PluginName = T.Text
 data IdePlugins = IdePlugins
   { ipMap :: Map.Map PluginId [UntaggedCommandDescriptor]
   } deriving (Show,Eq,Generic)
+instance ToSchema IdePlugins
+
+-- ---------------------------------------------------------------------
+
+-- | A position in a source file
+-- type Pos = (Line,Col)
+data Pos = Pos { line::Line, col:: Col} deriving (Generic,Show,Read,Eq,Ord)
+-- NOTE: Pos needs to be defined in record syntac otherwise the generically
+--       derived swagger schema does not work properly
+instance ToSchema Pos
+
+unPos :: Pos -> (Int,Int)
+unPos (Pos (Line l) (Col c)) = (l,c)
+
+toPos :: (Int,Int) -> Pos
+toPos (l,c) = Pos (Line l) (Col c)
+
+newtype Line = Line Int deriving (Generic,Show,Eq,Read,Ord,Enum,Real)
+instance ToSchema Line
+
+newtype Col  = Col  Int deriving (Generic,Show,Eq,Read,Ord,Enum,Real)
+instance ToSchema Col;
 
-type Pos = (Int,Int)
+unLine :: Line -> Int
+unLine (Line l) = l
+
+unCol :: Col -> Int
+unCol (Col c) = c
+
+deriving instance Num Line
+deriving instance Num Col
+
+deriving instance Integral Line
+deriving instance Integral Col
+
+
+instance Bounded Line where
+  minBound = 0
+  maxBound = 1000000
+
+instance Bounded Col where
+  minBound = 0
+  maxBound = 100000
+
```

Comment:
The swagger spec spits out bounds, which look horrible for lines and cols from -2^63 to +2^63-1.

But I still need to modify the swagger schema gen to actually use the bounds.

The upper bounds are to cope with some of the auto-gen'd source out there, which gets pretty extreme.